Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/CertificateStateHelper.kt

```diff
@@ -130,11 +136,107 @@ fun IncludeCertificateQrcodeCardBinding.bindValidityViews(
             startValidationCheckButton.isVisible = isPersonDetails
         }
     }
+}
+
+@Suppress("LongParameterList", "ComplexMethod")
+fun PersonOverviewItemBinding.setUIState(
+    primaryCertificate: CwaCovidCertificate,
+    secondaryCertificate: CwaCovidCertificate? = null,
+    colorShade: PersonColorShade,
+    statusBadgeText: Int = 0,
+    badgeCount: Int = 0,
+    onCovPassInfoAction: () -> Unit
+) {
+    val valid = primaryCertificate.isValid
+    val context = root.context
+    val color = when {
+        primaryCertificate.isValid -> colorShade
+        else -> PersonColorShade.COLOR_INVALID
+    }
+    backgroundImage.setImageResource(color.background)
+    starsImage.setImageDrawable(starsDrawable(context, color))
+    name.text = primaryCertificate.fullName
+
+    qrCodeCard.apply {
+        image.loadAny(primaryCertificate.getValidQrCode(Locale.getDefault().language)) {
+            crossfade(true)
+            loadingView(qrCodeCard.image, qrCodeCard.progressBar)
+        }
+        statusText.isVisible = statusBadgeText != 0
+        statusBadge.isVisible = statusBadgeText != 0
+        if (statusBadgeText != 0) {
+            statusBadge.text = context.getString(statusBadgeText)
+        }
+        covpassInfoTitle.isVisible = valid
+        covpassInfoButton.isVisible = valid
+        covpassInfoButton.setOnClickListener { onCovPassInfoAction() }
+        invalidOverlay.isGone = valid
+        image.isEnabled = valid
+        certificateToggleGroup.isVisible = secondaryCertificate != null
+        primaryCertificateButton.typeface = Typeface.DEFAULT_BOLD
+        secondaryCertificateButton.typeface = Typeface.DEFAULT
+        certificateToggleGroup.setOnCheckedChangeListener { checkedId ->
+            when (checkedId) {
+                R.id.primary_certificate_button -> {
+                    image.loadAny(primaryCertificate.getValidQrCode(Locale.getDefault().language)) {
+                        crossfade(true)
+                        loadingView(qrCodeCard.image, qrCodeCard.progressBar)
+                    }
+                    primaryCertificateButton.typeface = Typeface.DEFAULT_BOLD
+                    secondaryCertificateButton.typeface = Typeface.DEFAULT
+                }
+                R.id.secondary_certificate_button -> {
+                    image.loadAny(secondaryCertificate?.getValidQrCode(Locale.getDefault().language)) {
+                        crossfade(true)
+                        loadingView(qrCodeCard.image, qrCodeCard.progressBar)
+                    }
+                    primaryCertificateButton.typeface = Typeface.DEFAULT
+                    secondaryCertificateButton.typeface = Typeface.DEFAULT_BOLD
+                }
+            }
+        }
+    }
```

Comment:
Please crack it into small chunks to make  it readable ? maybe related view logic in one function each ?

Reply:
Addressed with latest changes